scripts/setup.sh

```diff
@@ -7,7 +7,7 @@ if [ -z "$NP2_MODULE_DIR" -o -z "$NP2_MODULE_PERMS" -o -z "$NP2_MODULE_OWNER" -o
 fi
 
 # avoid problems with sudo path
-SYSREPOCTL=`su -c "which sysrepoctl" $USER`
+SYSREPOCTL=`su -c "PATH=$PATH which sysrepoctl" $USER`
```

Comment:
`PATH` is passed **from the caller context** the same way `USER` is passed on the same line. Having `$PATH` inside quotes doesn't change that. The way you would set PATH to itself would be to either put `$PATH` inside single quotes like so `PATH='$PATH'` or escaping the dollar sign like so `PATH=\$PATH`.

Running the following line is a nice example:
```
su -c "echo original PATH is $PATH; echo new PATH is \$PATH" $USER
```
To me it shows:
```
original PATH is /home/andrei.pavel/.gem/ruby/2.5.0/bin:/home/andrei.pavel/.gem/ruby/2.6.0/bin:/home/andrei.pavel/.gem/ruby/2.7.0/bin:/home/andrei.pavel/.local/bin:/home/andrei.pavel/.cargo/bin:/usr/local/bin:/usr/local/sbin:/usr/share/gcc-10/python:/opt/sysrepo/bin:/opt/Netopeer2/bin:/opt/libyang/bin:/opt/flutter/bin:/opt/kea-tools/scripts:/opt/kea/sbin:/opt/kea/bin:/opt/cassandra/bin:/opt/build2/bin:/home/andrei.pavel/go/bin:/home/andrei.pavel/.pub-cache/bin:/home/andrei.pavel/.npm-global/bin:/home/andrei.pavel/.local/pipx/venvs/pip/bin:/usr/lib/go/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
new PATH is /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
```

There is an argument to be made here for **adding to** and not **replacing** the former PATH of the new `su` session which you would do with `PATH=\$PATH:$PATH` but I'm not a fan of it.

This PATH thing could also be replaced with adding `-l` or `--login` to the `su` command in which case `PATH` is passed automatically.

Reply:
Oh, okay, I understand now. The whole point of using `su` and `which` was to get the absolute path to *sysrepoctl* using the logged user `PATH` because `sudo` often uses a restricted `PATH` and *sysrepoctl* is not found. Based on this it seems `PATH` did not even got changed the way it is now but it did solve the problem I had on some distribution, which confuses me. But anyway, please use `-l` instead, it seems like the most straightforward solution, thanks.